PhysicsTools/NanoAOD/python/jetsAK4_Puppi_cff.py

```diff
@@ -215,12 +215,96 @@ def nanoAOD_addDeepInfoAK4(process,addParticleNet,addRobustParTAK4=False,addUnif
 jetPuppiTable.variables.muonSubtrDeltaPhi = Var("userFloat('muonSubtrRawPhi') - phi()",float,doc="muon-subtracted raw phi - phi",precision=10)
 
 jetPuppiForMETTask =  cms.Task(basicJetsPuppiForMetForT1METNano,corrT1METJetPuppiTable)
-
-#before cross linking
 jetPuppiUserDataTask = cms.Task(hfJetPuppiShowerShapeforNanoAOD)
-
-#before cross linking
 jetPuppiTask = cms.Task(jetPuppiCorrFactorsNano,updatedJetsPuppi,jetPuppiUserDataTask,updatedJetsPuppiWithUserData,finalJetsPuppi)
-
-#after cross linkining
 jetPuppiTablesTask = cms.Task(jetPuppiTable)
+
+from Configuration.Eras.Modifier_fastSim_cff import fastSim
+from PhysicsTools.NanoAOD.common_cff import ExtVar
+
+def nanoAOD_refineFastSim_puppiJet(process):
+    process.puppiJetRefineNN = cms.EDProducer(
+        "JetBaseMVAValueMapProducer",
+        backend             = cms.string("ONNX"),
+        batch_eval          = cms.bool(True),
+        disableONNXGraphOpt = cms.bool(True),
+        src                 = cms.InputTag("updatedJetsPuppi"),  # <<< HERE
+        weightFile = cms.FileInPath("PhysicsTools/NanoAOD/data/model_refinement_regression_31July_long_opset11.onnx"),
```

Comment:
@sbein -- if you change the `weightFile` path to `PhysicsTools/NanoAOD/data/fastSimPuppiJetRefineNN_31July2025.onnx`, we can test both PRs together. Up to you...

Reply:
Ah right, forgot to propagate this last-minute file name change. It's in now - testing both together would be great.